syntax = "proto3";

package pb;

service Monitor {
  rpc GetMonitor(Request) returns (Reponse);
}

message Request {
  
}

message Reponse {
  Cpu cpu = 1;
  Mem mem = 2;
  repeated Net net = 3;
  repeated Proc proc = 4;
}

message Cpu {
  string name = 1;
  int64 core_num = 2;
  float cpu_hz = 3;
  repeated float cpu_s = 4;
}

message Mem {
  int64 total = 1;
  int64 free = 2;
  int64 available = 3;
  int64 cached = 4;
  int64 swap_total = 5;
  int64 swap_free = 6;
}

message Net {
  string name = 1;
  int64 upload = 2;
  int64 upload_s = 3;
  int64 download = 4;
  int64 download_s = 5;
}

message Proc {
  string name = 1;
  string usr = 2;
  string state = 3;
  int64 pid = 4;
  int64 mem = 5;
  int64 thread_num = 6;
  float cpu_s = 7;
}
